core/sdk/src/api/observability/log_signal.rs

```diff
@@ -187,7 +187,7 @@ impl<T: std::fmt::Display + ContextAsJson> LogSignal<T> {
         }
     }
     pub fn emit_debug(&self) {
-        log::debug!("{}", serde_json::Value::from(self));
```

Comment:
@brendanobra : This was the change I recommended, but I thought you said you didn't want to do it this way. Change of heart or just a misunderstanding?